include/deal.II/sundials/n_vector.templates.h

```diff
@@ -570,8 +570,7 @@ namespace SUNDIALS
         (void)v;
         return nullptr;
 #  else
-        // We need to cast away const here, as SUNDIALS demands a pure `void *`.
-        return &(const_cast<MPI_Comm &>(get_communicator<VectorType>(v)));
+        return reinterpret_cast<void *>(get_communicator<VectorType>(v));
```

Comment:
@luca-heltai This is the only change I'm not 100% sure about because I don't know anything about SUNDIALS. All the sundials tests pass locally on my workstation

Reply:
Isn't here a difference between the two codes?
Now you are interpreting *the communicator* as a void pointer, while before we were returning the address of the referenced communicator (and interpreting automatically as a void *). I suspect this may not be the way SUNDIALS intend it to work, but I'm not familiar with this interface, as I did not write the `n_vector` deal.II wrapper.

Reply:
See https://github.com/dealii/dealii/pull/12921 and the discussion in https://github.com/dealii/dealii/issues/12914